scripts/select-random-player/select_random_player.lua

```diff
@@ -0,0 +1,30 @@
+local BE_SPECIAL = "Traitor"
+
+function selectRandomPlayer(be_special)
+
+    local players = dm.global_vars:get_var("GLOB"):get_var("alive_player_list")
+    local playerTable = {}
+
+    for _, player in players do
+        local client = player:get_var("client")
+        if not client then continue end
+        local be_special_client = client:get_var("prefs"):get_var("be_special")
+        for _, special in be_special_client do
+            if special == be_special then
+                table.insert(playerTable, player)
+                break
```

Comment:
I don't understand. We're only giving them the role if they already have the role?

Reply:
If you want to avoid giving them the role a second time, then this should be `table.find(be_special_client, be_special) == nil` for checking if it's already here

Reply:
This is checking their prefs to see if they want to be the role in the first place or not.

Reply:
I'll replace it with table.find though, didn't realise this function existed though I could've looked it up

Reply:
Oh okay, makes sense